docs/docsite/rst/conf.py

```diff
@@ -369,20 +419,35 @@
 # If false, no module index is generated.
 # latex_use_modindex = True
 
-autoclass_content = 'both'
+autoclass_content = "both"
 
 # Note:  Our strategy for intersphinx mappings is to have the upstream build location as the
-# canonical source. 
+# canonical source.
 intersphinx_mapping = {
-    'python': ('https://docs.python.org/2/', None),
-    'python3': ('https://docs.python.org/3/', None),
-    'jinja2': ('http://jinja.palletsprojects.com/', None),
-    'ansible_2_9': ('https://docs.ansible.com/ansible/2.9/', None),
-    'ansible_11': ('https://docs.ansible.com/ansible/11/', None),
+    "python": ("https://docs.python.org/2/", None),
+    "python3": ("https://docs.python.org/3/", None),
+    "jinja2": ("http://jinja.palletsprojects.com/", None),
+    "ansible_2_9": ("https://docs.ansible.com/ansible/2.9/", None),
+    "ansible_11": ("https://docs.ansible.com/ansible/11/", None),
 }
 
 # linckchecker settings
-linkcheck_ignore = [
-]
+linkcheck_ignore = []
 linkcheck_workers = 25
 # linkcheck_anchors = False
+
+redirects = {}
+if tags.has("core"):
+    redirects_file = Path("./_data/pages.yaml")
+    if redirects_file.exists():
+        with redirects_file.open(mode="r", encoding="utf-8") as f:
+            redirects = yaml.safe_load(f)
+elif tags.has("ansible"):
+    data_dir = Path("./_data")
+    if data_dir.exists():
+        redirects = {}
+        for yaml_file in data_dir.glob("*.yaml"):
+            with yaml_file.open(mode="r", encoding="utf-8") as f:
+                file_redirects = yaml.safe_load(f)
+                if file_redirects:
+                    redirects.update(file_redirects)
```

Comment:
I formatted the entire `conf.py` file, which might not be something we want to do for this PR because it makes it a little hard to see exactly what changed. I can revert that. Although I think formatting at some point would be a good idea.

Also not sure if this is too much cowbell or what. I mean initializing redirects and checking that things exist. Maybe this could be a lot simpler with fewer lines? I'd appreciate feedback on that please.